compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -1522,7 +1522,18 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         // as potential fallback candidates (#142006). To temporarily mask that issue, let's not
         // select at all if there are no early inherent candidates.
```

Comment:
This comment needs to be moved above the `Type` branch then, otherwise it no longer makes sense